awx/ui/client/test/unit/side-nav-item.spec.js

```diff
@@ -0,0 +1,62 @@
+describe('Components | Side Nav Item', () => {
+    let $compile;
+    let $rootScope;
+    let element;
+    let scope;
+
+    beforeEach(() => {
+        angular.mock.module('gettext');
+        angular.mock.module('I18N');
+        angular.mock.module('ui.router');
+        angular.mock.module('at.lib.services')
+        angular.mock.module('at.lib.components')
+    });
+
+    beforeEach(angular.mock.inject((_$compile_, _$rootScope_) => {
+        $compile = _$compile_;
+        $rootScope = _$rootScope_;
+        scope = $rootScope.$new();
+
+        element = angular.element('<at-layout><at-side-nav><at-side-nav-item icon-class="fa-tachometer" route="dashboard" name="DASHBOARD"></at-side-nav-item></at-layout></at-side-nav><');
+        element = $compile(element)(scope);
+        scope.name = 'dashboard';
+        scope.$digest();
+    }));
+
+    describe('Side Nav Item Controller', () => {
+        let LayoutCtrl;
+        let SideNavItem;
+        let SideNavItemCtrl;
+
+        beforeEach(() => {
+            LayoutCtrl = element.controller('atLayout');
+            SideNavItem = angular.element(element[0].querySelector('at-side-nav-item'));
+            SideNavItemCtrl = SideNavItem.controller('atSideNavItem');
+        });
+
+        it('layoutVm.currentState watcher should assign isRoute', () => {
+            let current = {'name': 'dashboard'};
+            $rootScope.$broadcast('$stateChangeSuccess', current);
+            scope.$digest();
+            expect(SideNavItemCtrl.isRoute).toBe(true);
+
+            current = {'name': 'inventories'};
+            $rootScope.$broadcast('$stateChangeSuccess', current);
+            scope.$digest();
+            expect(SideNavItemCtrl.isRoute).toBe(false);
+        });
+
+        it('go() should call $state.go()', angular.mock.inject((_$state_) => {
```

Comment:
I hadn't seen the use of angular.mock.inject within an `it` test function's callback, but I like that it is explicitly allowing you to spy on the thing you care about for the particular test.  Neat stuff!